lib/net/http.rb

```diff
@@ -1180,7 +1181,8 @@ def proxy_port
     # The username of the proxy server, if one is configured.
     def proxy_user
       if ENVIRONMENT_VARIABLE_IS_MULTIUSER_SAFE && @proxy_from_env
-        proxy_uri&.user
+        user = proxy_uri&.user
+        CGI::unescape(user) if user
```

Comment:
Related to https://github.com/ruby/net-http/pull/5#discussion_r571528379, I think they are referring to move the `require 'cgi/util'` L24 inside the conditions:
```suggestion
        user = proxy_uri&.user
        if user
          require 'cgi/util'
          CGI::unescape(user)
        end
```

Reply:
An alternative to this is to extract a private method and re-use it in both places:
```ruby
unescape(user) if user

private

def unescape(string)
  require 'cgi/util'
  CGI::unescape(string)
end
```

Reply:
> Related to [#5 (comment)](https://github.com/ruby/net-http/pull/5#discussion_r571528379), I think they are referring to move the `require 'cgi/util'` L24 inside the conditions:

I agree on this.